src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.cs

```diff
@@ -16,12 +16,15 @@ namespace Microsoft.Extensions.Logging
     [UnsupportedOSPlatform("browser")]
     public static class ConsoleLoggerExtensions
     {
+        internal const string RequiresDynamicCodeMessage = "Binding TOptions to configuration values may require generating dynamic code at runtime.";
         internal const string TrimmingRequiresUnreferencedCodeMessage = "TOptions's dependent types may have their members trimmed. Ensure all required members are preserved.";
 
         /// <summary>
         /// Adds a console logger named 'Console' to the factory.
         /// </summary>
         /// <param name="builder">The <see cref="ILoggingBuilder"/> to use.</param>
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
+            Justification = "AddConsoleFormatter and RegisterProviderOptions are only called with Options types which only have simple properties.")]
```

Comment:
Similar comments as @MichalStrehovsky around reference type being safe with RDC problematic APIs like MakeGenericXXX and not with Value Types. Not sure if these option types are preserved elsewhere to warrant suppress

Reply:
See the comment [above](https://github.com/dotnet/runtime/pull/73737#discussion_r943980138). The options' properties are being preserved because we annotated the `AddConsoleFormatter` and `RegisterProviderOptions` methods with `DynamicallyAccessedMemberTypes.All`, which preserves the top level properties. However, there are nested properties which aren't being preserved - which I logged https://github.com/dotnet/runtime/issues/73822 to address.